token.go

```diff
@@ -959,7 +961,7 @@ func (t tokenProcessor) nextToken() (tokenStruct, error) {
 		}
 		// we did not get cancellation confirmation, something is not
 		// right, this connection is not usable anymore
-		return nil, errors.New("did not get cancellation confirmation from the server")
+		return nil, ErrorCancelConfirmation
```

Comment:
>ErrorCancelConfirmation [](http://example.com/codeflow?start=14&length=23)

could this just return an instance of ServerError instead? That type is already handled to mark the connection as bad.
Creating new one-off error instances for specific situations doesn't seem scalable. What if there's a new scenario with the same requirement later on?

Reply:
I was thinking of this but i was not sure if it is a ServerError but yes, i will do it in the way if it is ServerError because its much more scalable